src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpSyntaxNode.cs

```diff
@@ -12,31 +12,44 @@ namespace Microsoft.DotNet.Interactive.HttpRequest;
 
 internal abstract class HttpSyntaxNode : HttpSyntaxNodeOrToken
 {
-    private TextSpan _span;
+    private TextSpan _fullSpan;
     private readonly List<HttpSyntaxNodeOrToken> _childNodesAndTokens = new();
+    private TextSpan _span;
+    private bool _isSignificant = false;
 
     private protected HttpSyntaxNode(
         SourceText sourceText,
         HttpSyntaxTree? syntaxTree) : base(sourceText, syntaxTree)
     {
     }
 
+    public override TextSpan FullSpan => _fullSpan;
+
+    public override bool IsSignificant => _isSignificant;
+
+    public override string Text => SourceText.ToString(Span);
+
     public override TextSpan Span => _span;
 
+    /// <summary>
+    /// Gets the text of the current node, including trivia.
+    /// </summary>
+    public string FullText => SourceText.ToString(FullSpan);
```

Comment:
It is a bit strange that `FullSpan` is available on the base `HttpSyntaxNodeOrToken` but `FullText` is not.

If we think that `FullSpan` and `FullText` are only meaningful for `HttpSyntaxNode`, perhaps we should also move `FullSpan` down to `HttpSyntaxNode`. Alternatively, we can move `FullText` up to the base type so that caller does not have to think about whether they are dealing a node or a token when they want to get its full text.

Reply:
This makes sense, but the meaning of the two is also different. On a token, `Span` really means the full span. `FullSpan` is used for things like `FindNode` and `GrowSpan`. Removing `HttpSyntaxNodeOrToken.FullSpan` results in needing to switch on the two types in a number of places.

Reply:
I see... Should we also move `FullText` to the base type then so that the API is consistent?